addons/mail/static/src/core/common/suggestion_service.js

```diff
@@ -191,35 +191,12 @@ export class SuggestionService {
     }
 
     getPartnerSuggestions(thread) {
-        let partners;
-        const isNonPublicChannel =
-            thread &&
-            (thread.channel_type === "group" ||
-                thread.channel_type === "chat" ||
-                (thread.channel_type === "channel" &&
-                    (thread.parent_channel_id || thread).group_public_id));
-        if (isNonPublicChannel) {
-            // Only return the channel members when in the context of a
-            // group restricted channel. Indeed, the message with the mention
-            // would be notified to the mentioned partner, so this prevents
-            // from inadvertently leaking the private message to the
-            // mentioned partner.
-            partners = thread.channel_member_ids
-                .map((member) => member.persona)
-                .filter((persona) => persona.type === "partner");
-            if (thread.channel_type === "channel") {
-                const group = (thread.parent_channel_id || thread).group_public_id;
-                partners = new Set([...partners, ...(group?.personas ?? [])]);
+        return Object.values(this.store.Persona.records).filter((persona) => {
+            if (thread?.model !== "discuss.channel" && persona.eq(this.store.odoobot)) {
```

Comment:
It's still not properly modular if "discuss.channel" is referenced outside of discuss, this should be overridden differently to not need this kind of condition here.